### PR TITLE
Dsd 201 radio buttons

### PR DESCRIPTION
**Pull request by kaplanmichael**
Fixes https://jira.nypl.org/browse/DSD-201

## **This PR does the following:**
- Moves radio button from Input component into its own new "Radio" component.
- Updates automated tests

### Front End Review:
- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/Input/Input.stories.tsx
Comment (EdwinGuzman) [inline src/components/Input/Input.stories.tsx:177]: Minor: since this was removed `useState` can be removed from the import. Github gives a nice message about it:

Screen Shot 2021-04-09 at 11 41 09 AM

File: src/components/Radio/Radio.test.tsx
Comment (EdwinGuzman) [inline src/components/Radio/Radio.test.tsx:57]: Can you add to this test that the `htmlFor` (`htmlfor` in case the react version doesn't get picked up by Enzyme) is the same as the `id` that is passed as a prop to `Radio`? Having a label is great, but unless it's associated to the `input` element then it's not useful.

File: src/components/Radio/Radio.test.tsx
Comment (EdwinGuzman) [inline src/components/Radio/Radio.test.tsx]: This is a bit weird to pass because it doesn't do anything to a radio type input.

File: src/components/Radio/Radio.tsx
Comment (EdwinGuzman) [inline src/components/Radio/Radio.tsx]: Wrap `<input>` in backticks. It renders weirdly without in storybook:

Screen Shot 2021-04-09 at 12 06 15 PM

File: src/components/Input/Input.stories.tsx
Comment (EdwinGuzman) [reply src/components/Input/Input.stories.tsx:177]: Looks like the Github Action test failed because of this linting issue as well.

File: src/components/Radio/Radio.stories.mdx
Comment (EdwinGuzman) [inline src/components/Radio/Radio.stories.mdx:14]: Not sure if this was intentional or settled in a different conversation, but this makes this component render in a different subtitle than the rest of the components. This is a nice change!

Screen Shot 2021-04-09 at 12 08 42 PM

**Comment by EdwinGuzman** [comment]
Looking good so far. There's an accessibility error for the main story:
Screen Shot 2021-04-09 at 12 10 09 PM

The `Radio` story is just missing an explicit `id` prop, similar to how you added `labelText: "Test Label"`. This is an easy miss and eventually, I think this could use a larger discussion -- the `id` prop is optional but breaks this accessibility check. Perhaps other components behave in a similar way and shouldn't  but this isn't necessary for this PR. Just adding the `id` to begin with and making a note of it in the story description is enough for now.

Are you going to add other stories for the error, disabled, checked, etc states or is that a separate PR?

File: src/components/Radio/Radio.stories.mdx
Comment (bigfishdesign13) [reply src/components/Radio/Radio.stories.mdx:14]: @EdwinGuzman 
This was intentional.  We will be moving the components into logic groups within Storybook and Form Elements is the first group we are addressing.

File: src/components/Input/Input.stories.tsx
Comment (kaplanmichael) [reply src/components/Input/Input.stories.tsx:177]: Fixed!  Thanks for catching that!

File: src/components/Radio/Radio.tsx
Comment (kaplanmichael) [reply src/components/Radio/Radio.tsx]: Fixed!

File: src/components/Radio/Radio.stories.mdx
Comment (kaplanmichael) [reply src/components/Radio/Radio.stories.mdx:14]: @EdwinGuzman Thanks!  Yeah, this grew out of a discussion with @bigfishdesign13 about how to better organize the various components.  I'm glad you like it; that means we're on the right track!

File: src/components/Radio/Radio.test.tsx
Comment (kaplanmichael) [reply src/components/Radio/Radio.test.tsx]: Fixed!

**Comment by kaplanmichael** [comment]
> Looking good so far. There's an accessibility error for the main story:
> Screen Shot 2021-04-09 at 12 10 09 PM
> 
> The `Radio` story is just missing an explicit `id` prop, similar to how you added `labelText: "Test Label"`. This is an easy miss and eventually, I think this could use a larger discussion -- the `id` prop is optional but breaks this accessibility check. Perhaps other components behave in a similar way and shouldn't but this isn't necessary for this PR. Just adding the `id` to begin with and making a note of it in the story description is enough for now.
> 
> Are you going to add other stories for the error, disabled, checked, etc states or is that a separate PR?

@bigfishdesign13 If its absence causes an accessibility violation, should the `id` prop instead be mandatory?

File: src/components/Radio/Radio.test.tsx
Comment (kaplanmichael) [reply src/components/Radio/Radio.test.tsx:57]: Done!

File: src/components/Radio/Radio.test.tsx
Comment (EdwinGuzman) [inline src/components/Radio/Radio.test.tsx]: Nit: you don't need the fragment element wrapper.

File: src/components/Radio/Radio.test.tsx
Comment (EdwinGuzman) [inline src/components/Radio/Radio.test.tsx:30]: 👍 
Can you add a check to an existing test to show that when `showLabel` is set to `true`, that the `aria-label` attribute does not get rendered?

**Comment by EdwinGuzman** [comment]
The update looks good! Just two minor comments.

Re: making the `id` prop mandatory, that's probably the safest way to go so developers always set one. Nbd but the `Radio` components in the `<Canvas>` should also have an id prop.

File: src/components/Radio/Radio.test.tsx
Comment (kaplanmichael) [reply src/components/Radio/Radio.test.tsx]: Good catch; fixed!

File: src/components/Radio/Radio.test.tsx
Comment (kaplanmichael) [reply src/components/Radio/Radio.test.tsx:30]: Done!

**Review by EdwinGuzman — APPROVED** [review]
Looks good to me but wait for one more review.

File: src/components/Radio/Radio.test.tsx
Comment (kaplanmichael) [reply src/components/Radio/Radio.test.tsx]: Fixed!

**Review by EdwinGuzman — APPROVED** [review]
Approving latest update.

File: src/components/Radio/Radio.stories.mdx
Comment (bigfishdesign13) [reply src/components/Radio/Radio.stories.mdx:14]: @kaplanmichael You'll eventually want to implement the auto-categorization method from DSD-220.

File: src/components/Radio/_Radio.scss
Comment (bigfishdesign13) [inline src/components/Radio/_Radio.scss:1]: We never talked directly about the philosophy behind the root class name for this component.  I feel this class name should be unique to this componet and style inheritance should come from the base styles.  Not from other components.  We should have a call about this.

File: src/components/Input/Input.test.tsx
Comment (bigfishdesign13) [inline src/components/Input/Input.test.tsx:218]: The Input component will eventually be deprecated, so I am not sure these changes are necessary.

**Review by bigfishdesign13 — DISMISSED** [review]

File: src/components/Radio/Radio.test.tsx
Comment (kaplanmichael) [reply src/components/Radio/Radio.test.tsx:30]: Done!

File: src/components/Radio/_Radio.scss
Comment (kaplanmichael) [reply src/components/Radio/_Radio.scss:1]: Per our conversation on Monday, 4/19, we will move these reorganization changes to a new JIRA story.

File: src/components/Input/Input.test.tsx
Comment (kaplanmichael) [reply src/components/Input/Input.test.tsx:218]: @bigfishdesign13 You're right. I did this so that I could run the test suite locally and not have it error.  Since the changes were already done a while back, I would recommend keeping as-is for now.  Thoughts?

**Pull request merged**
